### PR TITLE
Updated subtree from protobuf

### PR DESCRIPTION
**Pull request by satvu**
Update protobuf to get the changes from [this commit](https://github.com/Azure/azure-functions-language-worker-protobuf/commit/0d3f1041c0be282906004adc98b304ee3c567501). This will provide us with gRPC messages needed to implement the capability to process a single payload of `FunctionLoadRequest` messages. 

### Issue describing the changes in this PR

Part of the solution for [this host issue](https://github.com/Azure/azure-functions-host/issues/7880).

### Pull request checklist

* [x] My changes **do not** require documentation changes
  * [ ] Otherwise: Documentation issue linked to PR
* [x] My changes **should not** be added to the release notes for the next release
  * [ ] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
  * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] I have added all required tests (Unit tests, E2E tests)

### Additional information

Additional PR information

### CODE REVIEW

File: src/DotNetWorker.Grpc/protobuf/src/proto/FunctionRpc.proto [hunk 1]
```diff
@@ -282,12 +296,15 @@ message FunctionsMetadataRequest {
...
+message FunctionMetadataResponse {
   // list of function indexing responses
-  repeated FunctionLoadRequest function_load_requests_results = 1;
+  repeated RpcFunctionMetadata function_metadata_results = 1;
 
   // status of overall metadata request
   StatusResult result = 2;
+
+  // if set to true then host will perform indexing
+  bool useDefaultMetadataIndexing = 3;
```
Comment (kshyju) [inline src/DotNetWorker.Grpc/protobuf/src/proto/FunctionRpc.proto:307]: nit: I see other props use `_` (ex: `function_metadata_results`) and this one use camelCase. Should we use the same format to be consistent?

File: src/DotNetWorker.Grpc/protobuf/src/proto/FunctionRpc.proto [hunk 2]
```diff
@@ -273,6 +281,12 @@ message RpcFunctionMetadata {
 
   // Raw binding info
   repeated string raw_bindings = 10;
+
+  // unique function identifier (avoid name collisions, facilitate reload case)
+  string function_id = 13;
+
+  // A flag indicating if managed dependency is enabled or not
+  bool managed_dependency_enabled = 14;
```
Comment (kshyju) [inline src/DotNetWorker.Grpc/protobuf/src/proto/FunctionRpc.proto:289]: This is unrelated to single load request change. Could you remind me what this one is for?

**Review by kshyju — APPROVED** [review]

File: src/DotNetWorker.Grpc/protobuf/src/proto/FunctionRpc.proto (hunk 1 above)
Comment (satvu) [reply src/DotNetWorker.Grpc/protobuf/src/proto/FunctionRpc.proto:307]: Good catch - this is a Stein change so fixing it won't break anything existing -- the change will have to come from the [protobuf repo](https://github.com/Azure/azure-functions-language-worker-protobuf) though, will talk to @surgupta-msft offline about this.

File: src/DotNetWorker.Grpc/protobuf/src/proto/FunctionRpc.proto (hunk 2 above)
Comment (satvu) [reply src/DotNetWorker.Grpc/protobuf/src/proto/FunctionRpc.proto:289]: From [this commit](https://github.com/Azure/azure-functions-language-worker-protobuf/commit/67ecc31fcf4ee44bd60c45d65fa1c854b8723ea8) in the protobuf repo ~21 hrs ago. I'm not familiar with the change, but I'm pretty sure it's Stein related.

Comment (satvu) [reply src/DotNetWorker.Grpc/protobuf/src/proto/FunctionRpc.proto:289]: Oh it is - it was part of some refactoring done to better work with legacy apps. These fields were moved from a different message to this one.

Comment (surgupta-msft) [reply src/DotNetWorker.Grpc/protobuf/src/proto/FunctionRpc.proto:289]: This is part of host fallback workitem. There were inconsistencies in host and protobuf repo. Yesterday, all these PRs fully resolved them.
